drivers/usb/device/usb_dc_stm32.c

```diff
@@ -60,34 +62,34 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
+#define DT_USB_RAM_SIZE			DT_PROP(DT_INST(0, st_stm32_otghs), ram_size)
+#define DT_USB_MAXIMUM_SPEED		DT_PROP(DT_INST(0, st_stm32_otghs), maximum_speed)
+#define DT_USB_CLOCK_BITS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_otghs), bits)
+#define DT_USB_CLOCK_BUS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_otghs), bus)
 #elif defined(DT_COMPAT_ST_STM32_OTGFS)
-#define DT_USB_BASE_ADDRESS		DT_INST_0_ST_STM32_OTGFS_BASE_ADDRESS
-#define DT_USB_IRQ			DT_INST_0_ST_STM32_OTGFS_IRQ_OTGFS
-#define DT_USB_IRQ_PRI		    DT_INST_0_ST_STM32_OTGFS_IRQ_OTGFS_PRIORITY
-#define DT_USB_NUM_BIDIR_ENDPOINTS  DT_INST_0_ST_STM32_OTGFS_NUM_BIDIR_ENDPOINTS
-#define DT_USB_RAM_SIZE			DT_INST_0_ST_STM32_OTGFS_RAM_SIZE
-#define DT_USB_MAXIMUM_SPEED		DT_INST_0_ST_STM32_OTGFS_MAXIMUM_SPEED
-#define DT_USB_CLOCK_BITS		DT_INST_0_ST_STM32_OTGFS_CLOCK_BITS
-#define DT_USB_CLOCK_BUS		DT_INST_0_ST_STM32_OTGFS_CLOCK_BUS
+#define DT_USB_BASE_ADDRESS		DT_REG_ADDR(DT_INST(0, st_stm32_otgfs))
+#define DT_USB_IRQ			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_otgfs), otgfs, irq)
+#define DT_USB_IRQ_PRI			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_otgfs), otgfs, priority)
+#define DT_USB_NUM_BIDIR_ENDPOINTS	DT_PROP(DT_INST(0, st_stm32_otgfs), num_bidir_endpoints)
+#define DT_USB_RAM_SIZE			DT_PROP(DT_INST(0, st_stm32_otgfs), ram_size)
+#define DT_USB_MAXIMUM_SPEED		DT_PROP(DT_INST(0, st_stm32_otgfs), maximum_speed)
+#define DT_USB_CLOCK_BITS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_otgfs), bits)
+#define DT_USB_CLOCK_BUS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_otgfs), bus)
 #elif defined(DT_COMPAT_ST_STM32_USB)
-#define DT_USB_BASE_ADDRESS		DT_INST_0_ST_STM32_USB_BASE_ADDRESS
-#define DT_USB_IRQ			DT_INST_0_ST_STM32_USB_IRQ_USB
-#define DT_USB_IRQ_PRI			DT_INST_0_ST_STM32_USB_IRQ_USB_PRIORITY
-#define DT_USB_NUM_BIDIR_ENDPOINTS   DT_INST_0_ST_STM32_USB_NUM_BIDIR_ENDPOINTS
-#define DT_USB_RAM_SIZE			DT_INST_0_ST_STM32_USB_RAM_SIZE
-#define DT_USB_MAXIMUM_SPEED		DT_INST_0_ST_STM32_USB_MAXIMUM_SPEED
-#define DT_USB_CLOCK_BITS		DT_INST_0_ST_STM32_USB_CLOCK_BITS
-#define DT_USB_CLOCK_BUS		DT_INST_0_ST_STM32_USB_CLOCK_BUS
-#ifdef DT_INST_0_ST_STM32_USB_ENABLE_PIN_REMAP
-#define DT_USB_ENABLE_PIN_REMAP		DT_INST_0_ST_STM32_USB_ENABLE_PIN_REMAP
+#define DT_USB_BASE_ADDRESS		DT_REG_ADDR(DT_INST(0, st_stm32_usb))
+#define DT_USB_IRQ			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_usb), usb, irq)
+#define DT_USB_IRQ_PRI			DT_IRQ_BY_NAME(DT_INST(0, st_stm32_usb), usb, priority)
+#define DT_USB_NUM_BIDIR_ENDPOINTS	DT_PROP(DT_INST(0, st_stm32_usb), num_bidir_endpoints)
+#define DT_USB_RAM_SIZE			DT_PROP(DT_INST(0, st_stm32_usb), ram_size)
+#define DT_USB_MAXIMUM_SPEED		DT_PROP(DT_INST(0, st_stm32_usb), maximum_speed)
+#define DT_USB_CLOCK_BITS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bits)
+#define DT_USB_CLOCK_BUS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bus)
+#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
+#define DT_USB_ENABLE_PIN_REMAP		DT_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
```

Comment:
Is that working ?
These compats are mutually exclusives, I'm not sure you'll get it work that way.

Reply:
Why not?  The specific compat (`st_stm32_usb`, `st_stm32_otgfs` or `st_stm32_otghs`) is explicit specified in the macro.

Reply:
ok, got it. Will take more time to review this more carefully.

Reply:
thanks